test/controllers/provisioning_templates_controller_test.rb

```diff
@@ -144,6 +144,18 @@ class ProvisioningTemplatesControllerTest < ActionController::TestCase
       get :build_pxe_default, {}, set_session_user
       assert_redirected_to provisioning_templates_path
     end
+
+    test "pxe menu's labels should be sorted by full hostname title" do
+      t1 = TemplateCombination.new :hostgroup => hostgroups(:inherited), :environment => environments(:production)
+      t1.provisioning_template = templates(:mystring2)
+      t2 = TemplateCombination.new :hostgroup => hostgroups(:nested), :environment => environments(:production)
+      t2.provisioning_template = templates(:mystring2)
+      t1.save
+      t2.save
+      ProxyAPI::TFTP.any_instance.expects(:create_default).with(regexp_matches(/^PXE.*/), has_entry(:menu, regexp_matches(/#{hostgroups(:nested).name}.*#{hostgroups(:inherited).name}/m))).returns(true).times(3)
```

Comment:
If you really intend to test the ordering, explicitly create host groups with names above that will exercise it - it's not clear from the fixtures that the names are meant to be sorted this way.